python/ray/rllib/models/multiagentfcnet.py

```diff
@@ -6,38 +6,47 @@
 from ray.rllib.models.action_dist import Reshaper
 
+# TODO(cathywu) support more network types
+MODEL_CONFIGS = {"FullyConnectedNetwork": FCN_CONFIGS,
+                 "TwoLevelFCNetwork": TFCN_CONFIGS}
+
 
 class MultiAgentFullyConnectedNetwork(Model):
-    """Multiagent fully connected network."""
-
-    def _init(self, inputs, num_outputs, options):
-
-        # Split the input and output tensors
-        input_shapes = options["custom_options"]["multiagent_obs_shapes"]
-        output_shapes = options["custom_options"]["multiagent_act_shapes"]
-        input_reshaper = Reshaper(input_shapes)
-        output_reshaper = Reshaper(output_shapes)
-        split_inputs = input_reshaper.split_tensor(inputs)
-        num_actions = output_reshaper.split_number(num_outputs)
-
-        custom_options = options["custom_options"]
-        hiddens = custom_options.get("multiagent_fcnet_hiddens",
-                                     [[256, 256]]*1)
-
-        # check for a shared model
-        shared_model = custom_options.get("multiagent_shared_model", 0)
-        reuse = tf.AUTO_REUSE if shared_model else False
-        outputs = []
-        for i in range(len(hiddens)):
-            with tf.variable_scope("multi{}".format(i), reuse=reuse):
-                sub_options = options.copy()
-                sub_options.update({"fcnet_hiddens": hiddens[i]})
-                # TODO(ev) make this support arbitrary networks
-                fcnet = FullyConnectedNetwork(
-                    split_inputs[i], int(num_actions[i]), sub_options)
-                output = fcnet.outputs
-                outputs.append(output)
-        overall_output = tf.concat(outputs, axis=1)
-        return overall_output, outputs
+  """Multiagent fully connected network."""
+
+  def _init(self, inputs, num_outputs, options):
+    # Split the input and output tensors
+    input_shapes = options["custom_options"]["multiagent_obs_shapes"]
+    output_shapes = options["custom_options"]["multiagent_act_shapes"]
+    input_reshaper = Reshaper(input_shapes)
+    output_reshaper = Reshaper(output_shapes)
+    split_inputs = input_reshaper.split_tensor(inputs)
+    num_actions = output_reshaper.split_number(num_outputs)
+
+    custom_options = options["custom_options"]
+    hiddens = custom_options.get("multiagent_fcnet_hiddens",
+                                 [[256, 256]] * 1)
+
+    network_cls = globals()[custom_options.get("network_type", "FullyConnectedNetwork")]
+    SUBMODEL_CONFIGS = MODEL_CONFIGS[custom_options.get("network_type", "FullyConnectedNetwork")]
+
+    # check for a shared model
+    shared_model = custom_options.get("multiagent_shared_model", 0)
+    reuse = tf.AUTO_REUSE if shared_model else False
+    outputs = []
+    for i in range(len(hiddens)):
+      with tf.variable_scope("multi{}".format(i), reuse=reuse):
+        sub_options = options.copy()
+        sub_options.update({"fcnet_hiddens": hiddens[i]})
```

Comment:
I think some code I wrote is missing here. Namely, the `SUBMODEL_CONFIGS` are not used to create the appropriate model type.